### ISSUE TITLE
Temporary lose of data for statistics

### ISSUE THREAD

**Issue by KillianKemps** [getgrav/grav-plugin-admin#769]
Hello,

It happens sometimes that I only get the statistics of today but not the previous days like if the data before has been deleted. However, it seems the data of the month has been kept.

![screenshot from 2016-09-08 23-14-54](https://cloud.githubusercontent.com/assets/6884630/18367154/00f39be8-761a-11e6-8422-d9397f2825ac.png)

In fact, while writing this issue, I got the idea that weeks starts on Thursdays for the Admin plugin (at least for my Grav installation) and therefore it resets the week's data on this day.

By the way, it was about 22:30 (UTC+2, but the server has perhaps another timezone, maybe the default UTC) when I saw that I've lost my week's statistics even though some hours before I had it all. So maybe it has been reset at midnight.

If this is the problem, then how could I **configure the day** when it starts a week?

And why not display the **last 7 days** statistics instead of breaking statistics in weeks?

Here are some more infos about my installation:
- Grav v1.1.4
- Admin plugin v1.2.1
- Server is a shared-hosting running on Ubuntu 14.04

To be sure, I forced Grav's timezone to UTC+2 and I will let the days pass to confirm if it happens every Thursday.

**Comment by rhukster** [comment]
Actually it goes back through your 'days' and collects the last 7:

```
    public function getWeeklyTotal()
    {
        if (!$this->daily_data) {
            $this->daily_data = $this->getData($this->daily_file);
        }

        $day = 0;
        $total = 0;
        foreach (array_reverse($this->daily_data) as $daily) {
            $total += $daily;
            $day++;
            if ($day == 7) {
                break;
            }
        }

        return $total;
    }
```

**Comment by rhukster** [comment]
Actually just testing this and it does look wrong.. investigating.

**Comment by KillianKemps** [comment]
I see, than it is not a problem of when the week starts.

Yes, here is the content of `logs/popularity/daily.json`, but one day has passed since my first post, so there is one data more.

```
{"08-09-2016":120,"09-09-2016":36}
```

Here is the screenshot related to this data:

![screenshot from 2016-09-09 09-59-04](https://cloud.githubusercontent.com/assets/6884630/18379663/ff71768c-7673-11e6-9239-976771a8c083.png)


So, the display of the statistics widget seems normal according to the data, but than it means something deleted my weekly data.

**Comment by rhukster** [comment]
No the update shouldn't effect anything in the logs.  I suspect your hosting provider has some automated scripts that are periodically clearing out / backing up anything in `logs/` folder. 

This is not something Grav would do.

**Comment by KillianKemps** [comment]
Okay, it happened again:

`logs/popularity/daily.json` :
```
{"09-09-2016":183}
```

![screenshot from 2016-09-09 15-32-06](https://cloud.githubusercontent.com/assets/6884630/18388590/7ff44ef0-76a2-11e6-9840-544ed81f4168.png)

It is quite weird because some other days I have the complete week without any problem.

I investigated if my hosting provider has set up an automated script, but it seems there is nothing like that.

**Comment by flaviocopes** [comment]
Quite strange! If the hosting had a script to clear logs/, it would also clear the month data

**Comment by rhukster** [comment]
Yah that's my hunch as `logs/` is a very common directory name.. i bet your other logs are getting cleared out too..

**Comment by KillianKemps** [comment]
Well, this is the content of `monthly.json`: 
```
{"09-2016":3563}
```
If we consider my first screenshot, if all logs of `logs/` are deleted, than the monthly statistics should be far less.

Also, the file `logs/grav.log` hasn't been modified since 2016-08-30 and has kept all logs since the first installation. Other files which are direct children of `logs/` seem also to have normal behavior.

So, if my hosting provider is deleting all files in `logs/` it has a weird way of doing it.

However, I don't know how should be the file `logs/popularity/totals.json`, but I have pages which only had one visit when I'm very sure they had much more before. 
What I understand is that the `totals.json` file is deleted at the same time as `daily.json` for an unknown reason at that would explain the partial lose of data.

**Comment by flaviocopes** [comment]
There is nothing in Grav clearing those logs. They are initialised once when they are first non-existing, then updated over time, but never deleted. I have logs lasting several months and if there was any functionality to delete them, it would have triggered as I try every possible thing in Grav :)

No idea what it's deleting that file contents at this point. You might want to disable Popularity from the Admin plugin settings and use Google Analytics instead.

**Comment by KillianKemps** [comment]
Since I updated to Grav v1.1.5 and plugins including the Admin panel to v1.2.2 it seems that I do not have the issue anymore. It's been one week without a problem.

I hope the issue is permanently resolved now even though I still have no idea what is the cause.

I'm also using another Analytics tool, but as people can use anti-trackers browser addons, it doesn't work with all the visitors. On the other hand, Grav's analytics counts too much visits because I think it also counts when bots are getting on the website.
So I like to have both tools to compare and to have maximum and minimum numbers.

**Comment by flaviocopes** [comment]
Oh good - not sure what's the cause (and the solution) as we have not seen this problem in other users.

Closing then, reopen if it happens again.

**Comment by KillianKemps** [comment]
Well, it finally happened again on Saturday 1st October. I did absolutely nothing, no update, no new post. I just logged in the admin interface to check the statistics to see that I've lost the week's statistics. Only some visits from Friday still remained. However, today I still have the stats from Saturday.

But, I won't reopen the issue, because it seems I'm the only one to have it. It's perhaps some bug with the shared-hosting as you said before.

Grav v1.1.5 - Admin v1.2.2
